openpype/hosts/maya/plugins/publish/extract_active_view_thumbnail.py

```diff
@@ -0,0 +1,50 @@
+import maya.api.OpenMaya as om
+import maya.api.OpenMayaUI as omui
+
+import pyblish.api
+import tempfile
+
+
+class ExtractActiveViewThumbnail(pyblish.api.InstancePlugin):
+    """Set instance thumbnail to a screengrab of current active viewport.
+
+    This makes it so that if an instance does not have a thumbnail set yet that
+    it will get a thumbnail of the currently active view at the time of
+    publishing as a fallback.
+
+    """
+    order = pyblish.api.ExtractorOrder + 0.49
+    label = "Active View Thumbnail"
+    families = ["workfile"]
+    hosts = ["maya"]
+
+    def process(self, instance):
+        thumbnail = instance.data.get("thumbnailPath")
```

Comment:
This might also need an `IS_HEADLESS` check to ignore this plugin if maya is in batch/headless mode.

Reply:
If you'd add `targets =["local"]` you should be assured it's in UI mode.

Reply:
Hmm, ... maybe. Not entirely - technically I can run a headless publish on my machine as a local target, right? I mean, I've done that just fine many times with e.g. maya.

Reply:
```suggestion
        if IS_HEADLESS:
            self.log.debug(
                "Skip extraction of active view thumbnail, due to being in"
                "headless mode."
            )
            return

        thumbnail = instance.data.get("thumbnailPath")
```